run/local-cluster.go

```diff
@@ -0,0 +1,402 @@
+	}
+	c := createRoach(client, vols, -1,
+		append([]string{"cert", "--certs=/certs", "create-node"}, nodes...)...)
+	defer remove(client, c)
+	start(client, nil, c)
+	wait(client, c)
+}
+
+func initCluster(client *docker.Client, vols *docker.Container) {
+	log.Printf("initializing cluster")
+	c := createRoach(client, vols, -1,
+		"init", "--stores=ssd="+data(0))
+	defer remove(client, c)
+	start(client, nil, c)
+	wait(client, c)
+}
+
+func startNode(client *docker.Client, vols, dns *docker.Container,
+	numNodes, i int) *docker.Container {
+	cmd := []string{
+		"start",
+		"--stores=ssd=" + data(i),
+		"--certs=/certs",
+		"--addr=" + node(i) + ":8080",
+		"--gossip=" + node(0) + ":8080",
+	}
+	c := createRoach(client, vols, i, cmd...)
+	start(client, dns, c)
+	c = inspect(client, c.ID)
+	c.Name = node(i)
+	ports := c.NetworkSettings.PortMappingAPI()
+	log.Printf("started %s: %s:%d", c.Name, dockerIP(), ports[0].PublicPort)
+	return c
+}
+
+func checkGossipNodes(client *http.Client, node *docker.Container) int {
+	mappings := node.NetworkSettings.PortMappingAPI()
+	addr := fmt.Sprintf("%s:%d", dockerIP(), mappings[0].PublicPort)
+
+	resp, err := client.Get("https://" + addr + "/_status/gossip")
+	if err != nil {
+		return 0
+	}
+	defer resp.Body.Close()
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return 0
+	}
+	var m map[string]interface{}
+	if err := json.Unmarshal(b, &m); err != nil {
+		return 0
+	}
+	count := 0
+	infos := m["infos"].(map[string]interface{})
+	for k := range infos {
+		if strings.HasPrefix(k, "node:") {
+			count++
+		}
+	}
+	return count
+}
+
+func checkCluster(nodes []*docker.Container, done chan struct{}) {
+	defer func() {
+		done <- struct{}{}
+	}()
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}}
+
+	log.Printf("waiting for complete gossip network of %d peerings",
+		len(nodes)*len(nodes))
+
+	for i := 0; i < 20; i++ {
+		time.Sleep(1 * time.Second)
+		found := 0
+		for j := 0; j < len(nodes); j++ {
+			found += checkGossipNodes(client, nodes[j])
+		}
+		fmt.Printf("%d ", found)
+		if found == len(nodes)*len(nodes) {
+			fmt.Printf("... all nodes verified in the cluster\n")
+			return
+		}
+	}
+
+	fmt.Printf("... failed to verify all nodes in cluster\n")
+}
+
+func main() {
+	log.SetFlags(log.Ltime)
+	flag.Parse()
+
+	client := newDockerClient()
+	dns := runDockerSpy(client)
+	defer kill(client, dns)
+	log.Printf("started %s: %s\n", dns.Name, dns.NetworkSettings.IPAddress)
+
+	vols := createVolumes(client, *numNodes)
+	defer remove(client, vols)
+	defer os.RemoveAll("certs")
+
+	createCACert(client, vols)
+	createNodeCerts(client, vols, *numNodes)
+	initCluster(client, vols)
+
+	var nodes []*docker.Container
+	for i := 0; i < *numNodes; i++ {
+		c := startNode(client, vols, dns, *numNodes, i)
+		defer kill(client, c)
+		nodes = append(nodes, c)
+	}
+
+	done := make(chan struct{}, 1)
+	go checkCluster(nodes, done)
```

Comment:
I think we should have two distinct modes: either run checkCluster or wait for an interrupt, but I don't think it makes sense to do both. checkCluster would just spew results into the logs with no clean way to indicate its results (and once we add more tests, the tests done by checkCluster could cause problematic interactions with whatever is calling local-cluster.